libfreerdp/common/settings.c

```diff
@@ -1333,62 +1333,144 @@ BOOL freerdp_settings_set_pointer_len(rdpSettings* settings, size_t id, const vo
-				return NULL;
+			max = freerdp_settings_get_uint32(settings, FreeRDP_MonitorDefArraySize);
+			if (offset >= max)
+				goto fail;
 			return &settings->MonitorDefArray[offset];
-
 		case FreeRDP_ChannelDefArray:
-			if (offset > freerdp_settings_get_uint32(settings, FreeRDP_ChannelDefArraySize))
-				return NULL;
+			max = freerdp_settings_get_uint32(settings, FreeRDP_ChannelDefArraySize);
+			if (offset >= max)
+				goto fail;
 			return &settings->ChannelDefArray[offset];
 		case FreeRDP_DeviceArray:
-			if (offset > freerdp_settings_get_uint32(settings, FreeRDP_DeviceArraySize))
-				return NULL;
+			max = freerdp_settings_get_uint32(settings, FreeRDP_DeviceArraySize);
+			if (offset >= max)
+				goto fail;
 			return &settings->DeviceArray[offset];
 		case FreeRDP_StaticChannelArray:
-			if (offset > freerdp_settings_get_uint32(settings, FreeRDP_StaticChannelArraySize))
-				return NULL;
+			max = freerdp_settings_get_uint32(settings, FreeRDP_StaticChannelArraySize);
+			if (offset >= max)
+				goto fail;
 			return settings->StaticChannelArray[offset];
 		case FreeRDP_DynamicChannelArray:
-			if (offset > freerdp_settings_get_uint32(settings, FreeRDP_DynamicChannelArraySize))
-				return NULL;
+			max = freerdp_settings_get_uint32(settings, FreeRDP_DynamicChannelArraySize);
+			if (offset >= max)
+				goto fail;
 			return settings->DynamicChannelArray[offset];
 		case FreeRDP_FragCache:
-			if (offset >= 1)
-				return NULL;
+			max = 1;
+			if (offset >= max)
+				goto fail;
 			return &settings->FragCache[offset];
 		case FreeRDP_GlyphCache:
-			if (offset >= 10)
-				return NULL;
+			max = 10;
+			if (offset >= max)
+				goto fail;
 			return &settings->GlyphCache[offset];
 		case FreeRDP_BitmapCacheV2CellInfo:
-			/* TODO: BitmapCacheV2NumCells should be limited to 4
-			if (offset > freerdp_settings_get_uint32(settings, FreeRDP_BitmapCacheV2NumCells))
-			    return NULL;
-			    */
-
+			max = freerdp_settings_get_uint32(settings, FreeRDP_BitmapCacheV2NumCells);
+			if (offset >= max)
+				goto fail;
 			return &settings->BitmapCacheV2CellInfo[offset];
 		case FreeRDP_ReceivedCapabilities:
-			if (offset > settings->ReceivedCapabilitiesSize)
-				return 0;
+			max = freerdp_settings_get_uint32(settings, FreeRDP_ReceivedCapabilitiesSize);
+			if (offset >= max)
+				goto fail;
 			return &settings->ReceivedCapabilities[offset];
+		case FreeRDP_TargetNetAddresses:
+			max = freerdp_settings_get_uint32(settings, FreeRDP_TargetNetAddressCount);
+			if (offset >= max)
+				goto fail;
+			return settings->TargetNetAddresses[offset];
+		case FreeRDP_TargetNetPorts:
+			max = freerdp_settings_get_uint32(settings, FreeRDP_TargetNetAddressCount);
+			if (offset >= max)
+				goto fail;
+			return settings->TargetNetPorts[offset];
+		case FreeRDP_ClientTimeZone:
+			max = 1;
+			if (offset >= max)
+				goto fail;
+			return settings->ClientTimeZone;
+		case FreeRDP_RdpServerCertificate:
+			max = 1;
```

Comment:
Shouldn't all the parameters that are checked against 1 (`FreeRDP_RdpServerCertificate`, `FreeRDP_RdpServerRsaKey`, and so on) return NULL and an invalid id log message since they are not arrays and should be queried using `freerdp_settings_get_pointer_writable`?

Reply:
depends on how you view it, did allow these as `array[1]` types instead.